### PR TITLE
remove var usage in lock header

### LINKED ISSUE #30: circuits: extra large r1cs file sizes
r1cs file size for http_lock_header and aes_gctr is too big, even though constraint count is not that large. Find ways to reduce it

### PR DESCRIPTION
**Pull request by lonerapier**
closes #30 

reduces r1cs file size for `http_lock_header` circuit from 1.93G to 43MB by removing var usage in `FirstStringMatch` template.

Note: this doesn't happen with all `vars`, as can be noted here, only happening for iterating `var` accessed again in a separate signal/var.

### CODE REVIEW

File: circuits/http/nivc/lock_header.circom [hunk 1]
```diff
@@ -101,13 +103,13 @@ template FirstStringMatch(dataLen, maxKeyLen) {
         stringMatch[idx] = IsEqualArray(maxKeyLen);
         stringMatch[idx].in[0] <== key;
         for (var key_idx = 0 ; key_idx < maxKeyLen ; key_idx++) {
-            isFirstMatchAndInsideBound[idx * maxKeyLen + key_idx] <== (1 - matched) * (1 - isKeyOutOfBounds[key_idx]);
+            isFirstMatchAndInsideBound[idx * maxKeyLen + key_idx] <== (1 - isMatched[idx]) * (1 - isKeyOutOfBounds[key_idx]);
```
Comment (lonerapier) [inline circuits/http/nivc/lock_header.circom:106]: circom doesn't like updating a signal using an already updating `var`

Comment (0xJepsen) [reply circuits/http/nivc/lock_header.circom:106]: Very strange

**Review by 0xJepsen — APPROVED** [review]
This is so counter intuitive but honestly amazing that it works and reduces size by so much.

**Comment by 0xJepsen** [comment]
Okay i don't want to merge this without running CI and rebuilding the right artifacts. But i do thing we should build artifacts with these new perf gains asap.

**Comment by 0xJepsen** [comment]
LFG, we spin green to win

**Pull request merged**